manifests/0000_30_cluster-api_15_allow-egress-controllers.yaml

```diff
@@ -0,0 +1,125 @@
+# This NetworkPolicy allows specific egress traffic required for cluster-api controllers
+# while maintaining security by only allowing necessary connections.
+# This policy is applied to CAPI controller pods (control-plane label).
+# 
+# The egress rules enable controllers to:
+# 1. Communicate with Kubernetes API server (port 443, 6443)
+# 2. Resolve DNS queries (port 53 TCP/UDP)
+# 3. Perform health checks and metrics collection (port 9440, 9443, 8443)
+# 4. Access external cloud provider APIs (port 80, 443 to 0.0.0.0/0)
+# 5. Communicate with other CAPI components in the same namespace
+# 6. Communicate with webhook services for validation
+#
+# This policy works with the deny-all policy to implement a secure default-deny model
+# where only explicitly allowed traffic is permitted for controller operations.
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  annotations:
+    exclude.release.openshift.io/internal-openshift-hosted: "true"
+    include.release.openshift.io/self-managed-high-availability: "true"
+    include.release.openshift.io/single-node-developer: "true"
+    release.openshift.io/feature-set: CustomNoUpgrade,TechPreviewNoUpgrade
+  name: allow-egress-controllers
+  namespace: openshift-cluster-api
+spec:
+  egress:
+  - ports:
+    - port: 443 # HTTPS
+      protocol: TCP
+    to:
+    - ipBlock:
+        cidr: 172.30.0.0/16
```

Comment:
What is this range?

Reply:
this is serviceIP ranges , I tried in multiple installations it was same .. I guess but it can differ if there are custom installation configs.. and might have been skipped by me.
```
miyadav@miyadav-thinkpadx1carbongen8:~$ oc get networks cluster -o jsonpath='{.status.serviceNetwork}'
["172.30.0.0/16"]
```